### PR TITLE
ClojureScript support

### PR DESCRIPTION
**Pull request by k7d**
* Everything that doesn't rely on JVM-specific stuff should be supported in ClojureScript now
* Universal Clojure/ClojureScript unit-tests (migrated from midje to clojure.test for compatibility)

@skazhy

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -15,6 +16,15 @@
                  [clj-yaml "0.4.0" :exclusions [org.yaml/snakeyaml]]
                  [org.yaml/snakeyaml "1.15"]
                  [commons-codec/commons-codec "1.10"]]
+  :aliases {"at" ["test-refresh"]
```
Comment (skazhy) [inline project.clj]: These should live in your `~/.lein/profiles.clj`.

**Comment by skazhy** [comment]
Will check this tomorrow AM.

File: project.clj (hunk 1 above)
Comment (k7d) [inline project.clj]: imho matter of preference

File: src/full/core/log.cljc [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            (map #(if (string? %)
+                   % (pr-str %)))
+            (string/join " ")))
+
+     (defn enable-log-print!
+       "Set *print-fn* to console.log"
+       []
+       (set! *print-newline* false)
+       (set! *print-fn* (fn [& args] (.log js/console (format-log args)))))
+     )
```
Comment (skazhy) [inline src/full/core/log.cljc]: 🚓

File: src/full/core/log.cljc [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        (let [context (org.slf4j.LoggerFactory/getILoggerFactory)]
+          (try
+            (let [configurator (ch.qos.logback.classic.joran.JoranConfigurator.)]
+              (.setContext configurator context)
+              (.reset context)
+              (.doConfigure configurator (as-url (check-config-file config-file))))
+            (catch Exception _))  ; StatusPrinter will handle this
+          (ch.qos.logback.core.util.StatusPrinter/printInCaseOfErrorsOrWarnings
+            context))))
+     ))
```
Comment (skazhy) [inline src/full/core/log.cljc]: 🚓

File: src/full/core/log.cljc [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+(defn do-info
+  "Evaluates all arguments and logs them with info loglevel. Returns the value
+   of the last argument."
+  [& args]
+    (info (string/join ", " args))
+    (last args))
+
+(defn do-debug
```
Comment (skazhy) [inline src/full/core/log.cljc]: Will both `do-info` and `do-debug` work in cljs? The `info` and `debug` are defined only for Clojure.

File: src/full/core/sugar.cljc [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+         (if (= (.position bb) (.limit bb))
+           byte-vector
+           (recur (conj byte-vector (.get bb))))))
+
+     (defn byte-buffer->hex-string [byte-buffer]
+       (->> (byte-buffer->byte-vector byte-buffer)
+            (into-array Byte/TYPE)
+            (Hex/encodeHex)
+            (string/join)))
+     ))
```
Comment (skazhy) [inline src/full/core/sugar.cljc]: 🚓

**Comment by skazhy** [comment]
Looks ok, apart from the formatting police. I'm assuming we're moving all tests to `clojure.test` for shared libraries?

File: project.clj (hunk 1 above)
Comment (skazhy) [inline project.clj]: It's fine to have things like this in private projects, but if someone outside of FC decides to use it and has `at` mapped to something else globally, lein will be complaining.

File: src/full/core/sugar.cljc [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
+     :cljs (letfn [(top-32-bits [] (.toString (int (/ (.getTime (js/Date.)) 1000)) 16))
+                   (f [] (.toString (rand-int 16) 16))
+                   (g [] (.toString  (bit-or 0x8 (bit-and 0x3 (rand-int 15))) 16))]
+             (string/join (concat (top-32-bits)
+                                  (repeatedly 4 f) "4"
+                                  (repeatedly 3 f)
+                                  (g) (repeatedly 3 f)
+                                  (repeatedly 12 f))))))
+
+#?(:clj (def ^{:deprecated "Use uuids instead"} uuid uuids))
```
Comment (skazhy) [inline src/full/core/sugar.cljc]: I've removed a handful of methods when moving this out as `0.10.0-SNAPSHOT`, I think we can kill this one as well.

**Pull request merged**